bundles/org.openhab.ui.cometvisu/src/main/java/org/openhab/ui/cometvisu/internal/backend/rest/MultipartRequestMap.java

```diff
@@ -0,0 +1,124 @@
+
+    public MultipartRequestMap(HttpServletRequest request) {
+        this(request, System.getProperty("java.io.tmpdir"));
+    }
+
+    public MultipartRequestMap(HttpServletRequest request, String tempLocation) {
+        super();
+        try {
+            this.tempLocation = tempLocation;
+
+            encoding = request.getCharacterEncoding();
+            if (encoding == null) {
+                try {
+                    encoding = DEFAULT_ENCODING;
+                    request.setCharacterEncoding(DEFAULT_ENCODING);
+                } catch (UnsupportedEncodingException e) {
+                    throw new IllegalStateException("Unsupported encoding: " + encoding, e);
+                }
+            }
+
+            for (Part part : request.getParts()) {
+                String fileName = part.getSubmittedFileName();
+                if (fileName == null) {
+                    putMulti(part.getName(), getValue(part));
+                } else {
+                    processFilePart(part, fileName);
+                }
+            }
+        } catch (IOException | ServletException e) {
+            throw new IllegalStateException("Failed to get request parts", e);
+        }
+    }
+
+    public String getStringParameter(String name) {
+        List<Object> list = get(name);
+        return (list != null) ? (String) get(name).get(0) : null;
+    }
+
+    public File getFileParameter(String name) {
+        List<Object> list = get(name);
+        return (list != null) ? (File) get(name).get(0) : null;
+    }
+
+    private void processFilePart(Part part, String fileName) throws IOException {
+        File tempFile = new File(tempLocation, fileName);
+        tempFile.createNewFile();
+        tempFile.deleteOnExit();
+
+        try (BufferedInputStream input = new BufferedInputStream(part.getInputStream(), 8192);
+                BufferedOutputStream output = new BufferedOutputStream(new FileOutputStream(tempFile), 8192);) {
+
+            byte[] buffer = new byte[8192];
+            for (int length = 0; ((length = input.read(buffer)) > 0);) {
+                output.write(buffer, 0, length);
+            }
+        }
+        part.delete();
+        putMulti(part.getName(), tempFile);
+    }
+
+    private String getValue(Part part) throws IOException {
+        BufferedReader reader = new BufferedReader(new InputStreamReader(part.getInputStream(), encoding));
+        StringBuilder value = new StringBuilder();
+        char[] buffer = new char[8192];
+        for (int length; (length = reader.read(buffer)) > 0;) {
+            value.append(buffer, 0, length);
+        }
+        return value.toString();
+    }
+
+    private <T> void putMulti(final String key, final T value) {
+        List<Object> values = super.get(key);
+
+        if (values == null) {
+            values = new ArrayList<>();
+            values.add(value);
+            put(key, values);
+        } else {
+            values.add(value);
+        }
+    }
+}
```

Comment:
Newline missing here, I think.

Reply:
Yes that was part of the [force push](https://github.com/openhab/openhab-webui/compare/defa1a9995d07c59f5516634fb3057eabe7d0c95..d17efb8bf127ac093ce4491cb668cee3cb0314ed). :-)